sdk-extensions/metric-incubator/src/main/java/io/opentelemetry/sdk/viewconfig/ViewConfig.java

```diff
@@ -0,0 +1,220 @@
+                getAsType(viewConfigSpecMap, "selector", Map.class), "selector is required");
+        Map<String, Object> viewSpecMap =
+            Objects.requireNonNull(
+                getAsType(viewConfigSpecMap, "view", Map.class), "view is required");
+
+        InstrumentType instrumentType =
+            Optional.ofNullable(getAsType(selectorSpecMap, "instrument_type", String.class))
+                .map(InstrumentType::valueOf)
+                .orElse(null);
+        List<String> attributeKeys =
+            Optional.ofNullable(
+                    ((List<Object>) getAsType(viewSpecMap, "attribute_keys", List.class)))
+                .map(objects -> objects.stream().map(String::valueOf).collect(toList()))
+                .orElse(null);
+
+        result.add(
+            ViewConfigSpecification.builder()
+                .selectorSpecification(
+                    SelectorSpecification.builder()
+                        .instrumentName(getAsType(selectorSpecMap, "instrument_name", String.class))
+                        .instrumentType(instrumentType)
+                        .meterName(getAsType(selectorSpecMap, "meter_name", String.class))
+                        .meterVersion(getAsType(selectorSpecMap, "meter_version", String.class))
+                        .meterSchemaUrl(
+                            getAsType(selectorSpecMap, "meter_schema_url", String.class))
+                        .build())
+                .viewSpecification(
+                    ViewSpecification.builder()
+                        .name(getAsType(viewSpecMap, "name", String.class))
+                        .description(getAsType(viewSpecMap, "description", String.class))
+                        .aggregation(getAsType(viewSpecMap, "aggregation", String.class))
+                        .attributeKeys(attributeKeys)
+                        .build())
+                .build());
+      }
+      return result;
+    } catch (IOException e) {
+      throw new ConfigurationException(
+          "An error occurred reading view config file:  " + viewConfigYamlFile.getAbsolutePath(),
+          e);
+    } catch (RuntimeException e) {
+      throw new ConfigurationException(
+          "Failed to parse view config file: " + viewConfigYamlFile.getAbsolutePath(), e);
+    }
+  }
+
+  @Nullable
+  @SuppressWarnings("unchecked")
+  private static <T> T getAsType(Map<String, Object> map, String key, Class<T> type) {
+    Object value = map.get(key);
+    if (value != null && !type.isInstance(value)) {
+      throw new IllegalStateException(
+          "Expected "
+              + key
+              + " to be type "
+              + type.getName()
+              + " but was "
+              + value.getClass().getName());
+    }
+    return (T) value;
+  }
+
+  // Visible for testing
+  static View toView(ViewSpecification viewSpec) {
+    ViewBuilder builder = View.builder();
+    Optional.ofNullable(viewSpec.getName()).ifPresent(builder::setName);
```

Comment:
Above `ofNullable` is providing some value through `map` I think and is ok but I would say `ifPresent` at least without some other mapping, is almost always an anti-pattern, null checks in if statements are simpler and keep the code easier to read

Reply:
Oh but its so many extra lines! I've gotten use to reading `Optional.ofNullable().ifPresent(..)` and don't mind it if lambdas can be used in `ifPresent()` (preferably method references), and when throughput is low so the additional allocation for optional is negligible. 

I get it though - not the style I prefer but good to have a convention and stick with it.